cmd/tools/codegen/helpers.go

```diff
@@ -0,0 +1,94 @@
+package codegen
+
+import (
+	"go/parser"
+	"go/token"
+	"io"
+	"log"
+	"os"
+	"path"
+	"strings"
+	"text/template"
+
+	"golang.org/x/text/cases"
+	"golang.org/x/text/language"
+)
+
+func GenerateToFile[T any](
+	generator func(io.Writer, T) error,
+	generatorArg T,
+	outPath string,
+	outFileName string,
+) {
+	filename := path.Join(outPath, outFileName+"_gen.go")
+	w, err := os.Create(filename)
+	FatalIfErr(err)
+	defer func() {
+		FatalIfErr(w.Close())
+		_, err = parser.ParseFile(token.NewFileSet(), filename, nil, parser.SkipObjectResolution)
+		FatalIfErr(err)
+	}()
+	err = generator(w, generatorArg)
+	FatalIfErr(err)
+}
+
+func GenerateTemplateToWriter(tmpl string, data any, w io.Writer) error {
+	t, err := template.New("code").Parse(tmpl)
+	if err != nil {
+		return err
+	}
+	return t.Execute(w, data)
+}
+
+func GenerateTemplateToFile(
+	tmpl string,
+	data any,
+	outPath string,
+	outFileName string,
+) {
+	GenerateToFile(func(w io.Writer, data any) error {
+		return GenerateTemplateToWriter(tmpl, data, w)
+	}, data, outPath, outFileName)
+}
+
+func FatalIfErr(err error) {
+	if err != nil {
+		//nolint:revive // calls to log.Fatal only in main() or init() functions (revive)
```

Comment:
explain why it's okay
```suggestion
		//nolint:revive // okay to call Fatal here since this is part of the build process, not the server
```